src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnection.java

```diff
@@ -87,4 +106,78 @@ public OracleDbConnection(Connection jdbcConnection, List<DbType> customDataType
     }
     return tables;
   }
+
+  @Override
+  public Struct createStruct(String typeName, Object[] attributes) throws SQLException {
+    return super.createStruct(typeName, attributes);
+  }
+
+  @Override
+  public Array createArrayOf(String typeName, Object[] values) throws SQLException {
+    if (getCreateArrayMethod() == null) {
+      return super.createArrayOf(typeName, values);
+    } else {
+      try {
+        resolveLobs(typeName, values, new CollectionTypeResolver(this));
+        return (Array) getCreateArrayMethod().invoke(getJdbcConnection(), typeName, values);
+      } catch (Exception e) {
+        throw new SQLException("Error creating ARRAY", e);
+      }
+    }
+  }
+
+  private Method getCreateArrayMethod() {
+    if (createArrayMethod == null) {
+      try {
+        createArrayMethod = getJdbcConnection().getClass().getMethod("createARRAY", String.class, Object.class);
+        createArrayMethod.setAccessible(true);
+      } catch (NoSuchMethodException e) {
+        // Ignore, will use the standard method
+      }
+    }
+    return createArrayMethod;
+  }
+
+  @Override
+  protected void resolveLobs(String typeName, Object[] attributes, TypeResolver typeResolver) throws SQLException {
+    Map<Integer, ResolvedDbType> dataTypes = getLobFieldsDataTypeInfo(typeResolver.resolveType(typeName));
+
+    for (Map.Entry entry : dataTypes.entrySet()) {
+      Integer index = (Integer) entry.getKey();
+      ResolvedDbType dataType = (ResolvedDbType) entry.getValue();
+      // In Oracle we do not have the data type for structs or arrays, as the
+      // the driver does not provide the getAttributes functionality
+      // in their DatabaseMetaData.
+      // It has to be taken into account that the data type depends on JDBC, so the
+      // driver is the unit responsible for the mapping and we do not have that information
+      // in the DB catalog. We resolve the lobs depending on the name only.
+      typeResolver.resolveLobs(attributes, index - 1, dataType.getName());
+    }
+  }
+
+  @Override
+  protected Map<Integer, ResolvedDbType> getLobFieldsDataTypeInfo(String typeName) throws SQLException {
+    Map<Integer, ResolvedDbType> dataTypes = new HashMap<>();
+
+    String owner = getOwnerFrom(typeName);
+    String type = getTypeSimpleName(typeName);
+
+    String query = QUERY_TYPE_ATTRS + (owner != null ? QUERY_OWNER_CONDITION : "");
+
+    try (PreparedStatement ps = this.prepareStatement(query)) {
+      ps.setString(1, type);
+      if (owner != null) {
+        ps.setString(2, owner);
+      }
+
+      try (ResultSet resultSet = ps.executeQuery()) {
+        while (resultSet.next()) {
+          ResolvedDbType resolvedDbType = new ResolvedDbType(UNKNOWN_DATA_TYPE, resultSet.getString(ATTR_TYPE_NAME_PARAM));
+          dataTypes.put(resultSet.getInt(ATTR_NO_PARAM), resolvedDbType);
+        }
+      }
+    }
+    return dataTypes;
```

Comment:
where are the resultSet and PrepareStatement closed?

Reply:
The try with resources statement ensures that each resource is closed at the end of the statement: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html